Service/src/routes/portfolio.py

```diff
@@ -3,53 +3,74 @@
 import simplejson as json
 
 from routes import stock
+from auth import auth
+from request_validator import validator
+from request_validator.schemas import portfolio_schema
 from util.utility import Utility
 from util.error_map import errors
 
 portfolio_api = Blueprint('portfolio_api', __name__)
 
+DEFAULT_BUYPOWER = 10000
 
-@portfolio_api.route('/api/portfolio', methods=['POST'])
+@portfolio_api.route('/api/portfolios', methods=['POST'])
+@auth.login_required
+@validator.validate_body(portfolio_schema.fields)
 def post_portfolio():
    body = request.get_json()
-   try:
-      result = PortfolioSchema().load(body)
-   except ValidationError as err:
-      return make_response(json.dumps(err.messages), 400)
+   buyPower = body.get('buyPower') or DEFAULT_BUYPOWER
 
-   now = datetime.now()
+   if 'inviteCode' in body:
```

Comment:
Do you mind explaining what this check is for? (or just add a comment explaining it)

Reply:
Maybe a 501 (Not Implemented) would be more appropriate, much like the Java's UnsupportedOperationException

Reply:
I looked online and found this to be the best alternative. 500 implies something broke and I don't want to imply that. 405 alerts the client that this is a properly handled issue.